ui/src/components/credentials/CredentialLinkQR.tsx

```diff
@@ -0,0 +1,310 @@
+            setShareCredentialQRCode({ error: response.error, status: "failed" });
+          }
+        }
+      }
+    },
+    [linkID, env]
+  );
+
+  useEffect(() => {
+    const { aborter } = makeRequestAbortable(createCredentialQR);
+
+    return aborter;
+  }, [createCredentialQR]);
+
+  useEffect(() => {
+    const checkCredentialQRCode = async () => {
+      if (isAsyncTaskDataAvailable(shareCredentialQRCode) && linkID) {
+        const response = await getCredentialLinkQRCode({
+          env,
+          linkID,
+          sessionID: shareCredentialQRCode.data.sessionID,
+        });
+
+        if (response.isSuccessful) {
+          if (response.data.status !== "pending") {
+            setCredentialQRCheck({ data: response.data, status: "successful" });
+
+            const { proofTypes } = shareCredentialQRCode.data.linkDetail;
+
+            if (proofTypes.includes("SparseMerkleTreeProof")) {
+              void message.info("Issuance process started");
+            }
+
+            if (proofTypes.includes("BJJSignature2021")) {
+              void message.success("Credential sent");
+            }
+          }
+        } else {
+          setCredentialQRCheck({ error: response.error, status: "failed" });
+
+          void message.error(response.error.message);
+        }
+      }
+    };
+
+    const checkQRCredentialStatusTimer = setInterval(() => {
+      if (
+        (isAsyncTaskDataAvailable(credentialQRCheck) &&
+          credentialQRCheck.data.status !== "pending") ||
+        hasAsyncTaskFailed(credentialQRCheck)
+      ) {
+        clearInterval(checkQRCredentialStatusTimer);
+      } else {
+        void checkCredentialQRCode();
+      }
+    }, POLLING_INTERVAL);
+
+    return () => clearInterval(checkQRCredentialStatusTimer);
+  }, [shareCredentialQRCode, linkID, credentialQRCheck, env]);
+
+  const onStartAgain = () => {
+    makeRequestAbortable(createCredentialQR);
+    setCredentialQRCheck({ status: "pending" });
+  };
+
+  const hasFailed = hasAsyncTaskFailed(shareCredentialQRCode)
+    ? shareCredentialQRCode.error
+    : hasAsyncTaskFailed(credentialQRCheck)
+    ? credentialQRCheck.error
+    : undefined;
+
+  if (hasFailed) {
+    if (hasFailed.status === 404) {
+      return (
+        <Space align="center" direction="vertical" size="large">
+          <Avatar className="avatar-color-error" icon={<QRIcon />} size={56} />
+
+          <Typography.Title level={2}>Claim link is invalid</Typography.Title>
+
+          <Typography.Text type="secondary">
+            In case you think this is an error, please contact the issuer of this claim.
+          </Typography.Text>
+        </Space>
+      );
+    } else if (hasFailed.status === 400) {
+      return (
+        <Space align="center" direction="vertical" size="large">
+          <Avatar className="avatar-color-error" icon={<AlertIcon />} size={56} />
+
+          <Typography.Title level={2}>QR code expired, start again</Typography.Title>
```

Comment:
I would make this more in line with our tone and say "The credential link has expired, please start again" ([thread](https://www.figma.com/file/BE3tUbvtSB2EqmJsybpeFC?node-id=5792:165929#427129353)).